aws-datastore/src/test/java/com/amplifyframework/datastore/syncengine/SyncProcessorTest.java

```diff
@@ -400,6 +496,48 @@ public void userProvidedErrorCallbackInvokedOnFailure() {
         assertEquals(1, errorHandlerCallCount);
     }
 
+    private void assertOperationTypeCounts(ModelSyncedEvent event,
+                                           int expectedAdd,
+                                           int expectedUpdate,
+                                           int expectedDelete) {
+        assertEquals(expectedAdd, event.getAdded());
+        assertEquals(expectedUpdate, event.getUpdated());
+        assertEquals(expectedDelete, event.getDeleted());
+    }
+
+    private static HubAccumulator createAccumulator(HubEventFilter eventFilter, int times) {
+        return HubAccumulator.create(HubChannel.DATASTORE, eventFilter, times);
+    }
+
+    private static HubEventFilter forEvent(DataStoreChannelEventName eventName) {
+        return hubEvent -> eventName.toString().equals(hubEvent.getName());
+    }
+
+    @SuppressWarnings("unchecked")
+    private HubEventFilter syncMetricsEmittedFor(Class<? extends Model>... models) {
+        List<String> modelNames = Observable.fromIterable(Arrays.asList(models))
+            .map(model -> model.getSimpleName())
+            .toList()
+            .blockingGet();
```

Comment:
We have this pattern pretty frequently, by now.

We could wrap it into a utility:
```java
List<String> modelNames = ForEach.inList(models, Model::getSimpleName);
```
```java
// In testutils, probskis.
public final class ForEach {
     private ForEach() {}

     <I, O> List<O> inList(List<I> in, Mapping<I, O> mapping) {
         final List<O> out = new ArrayList<>();
         for (I item : in) {
             out.add(mapping.apply(item));
         }
         return Immutable.of(out);
     }

     @FunctionalInterface
     interface Mapping<I, O> {
         O apply(I in);
     }
}
```
Software is so cool. Sigh.

Reply:
I like it :)